source/develop/release-management/features/network/openstack_identity_api_v3.html.md

```diff
@@ -0,0 +1,304 @@
+
+## Limitations
+
+OpenStack Identity API provides a catalog of multiple endpoints, which
+provides the URL to access the OpenStack service. This feature does change
+the current implementation in oVirt Engine, which requires the user to specify
+this URL manually.
+
+## Benefit to oVirt
+
+Support for OpenStack Identity API v3 in oVirt Engine is required to enable 
+access to services from [OpenStack instances which does not support Identity
+API v2.0, especially the OpenStack releases since Qeens.][12]
+
+## User Experience
+
+### Webadmin
+
+The "Add Provider" popup has to provide inputs for the new properties user
+domain name, the project name and the domain name of the project.
+To avoid inconsistency between the authentication URL and the API version, the
+authentication URL is built from the protocol, host name, API port and version.
+
+![add a new provider](../../../../images/features/network/openstack_idendity_api-add-dialog-provider.png)
+
+### Engine-Setup
+
+The ovirt-provider-ovn external provider which is configured by engine-setup is
+not modified and continues to use OpenStack Identity API v2.0, because of the
+Engine and the OVN provider has to support OpenStack Identity API v2.0 for the
+sake of backward compatibility anyway and switching to OpenStack Identity API
+v3 would be no improvement but introduce the risk of new implementation bugs.
+
+## REST-API
+The REST-API is extended to handle the user domain name, the project name and
+the domain name of the project like the following:
+~~~~xml
+<openstack_network_provider href="/ovirt-engine/api/openstacknetworkproviders/123" id="123">
+  <actions>
+    <link 
+      href="/ovirt-engine/api/openstacknetworkproviders/123/testconnectivity"
+      rel="testconnectivity" />
+    <link
+      href="/ovirt-engine/api/openstacknetworkproviders/123/importcertificates"
+      rel="importcertificates"/>
+  </actions>
+  <name>OpenStack Neutron</name>
+  <description>network provider for OpenStack</description>
+  <link
+    href="/ovirt-engine/api/openstacknetworkproviders/123/networks"
+    rel="networks"/>
+  <link
+    href="/ovirt-engine/api/openstacknetworkproviders/123/certificates"
+    rel="certificates"/>
+  <url>http://192.168.122.104:9696</url>
+  <auto_sync>true</auto_sync>
+  <read_only>false</read_only>
+  <type>external</type>
+  <unmanaged>false</unmanaged>
+  <requires_authentication>true</requires_authentication>
+  <authentication_url>http://192.168.122.104:5000/v3</authentication_url>
+  <username>admin@internal</username>
+  <project_name>admin</project_name><!-- Added -->
+  <user_domain_name>Default</user_domain_name><!-- Added -->
+  <project_domain_name>Default</project_domain_name><!-- Added -->
+</openstack_network_provider>
+~~~~
+
+## Installation/Upgrade
+
+No effect on new installations of oVirt Engine.
+On upgrade, it has to be ensured that the authentication URL has the expected
+format to be splitted like proposed in [Webadmin](#webadmin).
```

Comment:
Dominic, is there something that the user actually has to do if upgrading to 4.2.6 (with support for v3) or will this be handled behind the scenes?

Reply:
No, the user has to execute engine-setup as described already in the upgrade guide.
engine-setup will fail if the authentication URL of any already configured provider has not the expected format.